apps/web-giddh/src/app/daybook/daybook.component.ts

```diff
@@ -376,58 +378,80 @@ export class DaybookComponent implements OnInit, OnDestroy {
+                                    let type = response?.queryString?.requestType === 'application/pdf' ? '.pdf' : '.xls';
+                                    saveAs(blob, 'Daybook' + type);
+                                }
+                            } else {
+                                this.toasterService.showSnackBar("error", response?.message);
+                            }
+                        });
+                    }
+                } else if (this.daybookExportRequestType === 'get') {
+                    this.daybookService.ExportDaybook(null, this.daybookQueryRequest).pipe(takeUntil(this.destroyed$)).subscribe(response => {
                         if (response?.status === 'success') {
                             if (response?.body?.type === "message") {
                                 this.toasterService.showSnackBar("success", response?.body?.file);
                             } else {
                                 let blob = this.generalService.base64ToBlob(response?.body?.file, response?.queryString?.requestType, 512);
                                 let type = response?.queryString?.requestType === 'application/pdf' ? '.pdf' : '.xls';
-                                saveAs(blob, 'Daybook' + type);
+                                saveAs(blob, 'response' + type);
                             }
                         } else {
                             this.toasterService.showSnackBar("error", response?.message);
                         }
                     });
                 }
-            } else if (this.daybookExportRequestType === 'get') {
-                this.daybookService.ExportDaybook(null, this.daybookQueryRequest).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            } else {
+                // for expanded option download
+                let exportBodyRequestObj: ExportBodyRequest = new ExportBodyRequest();
+                exportBodyRequestObj.from = this.daybookQueryRequest.from;
+                exportBodyRequestObj.to = this.daybookQueryRequest.to;
+                exportBodyRequestObj.fileType = this.fileTypeExtension;
+                exportBodyRequestObj.exportType = "ENTRIES_EXPORT";
+                let branchUniqueName = this.generalService.currentBranchUniqueName ? this.generalService.currentBranchUniqueName : this.currentBranch ? this.currentBranch?.uniqueName : "";
+                this.daybookService.exportDaybookExpandedPost(exportBodyRequestObj, branchUniqueName).pipe(takeUntil(this.destroyed$)).subscribe(response => {
                     if (response?.status === 'success') {
                         if (response?.body?.type === "message") {
-                            this.toasterService.showSnackBar("success", response?.body?.file);
+                            this.toasterService.showSnackBar("success", response?.body?.encodedData);
```

Comment:
@nisha-walkover are you receiving message in encodedData key???

Reply:
yes sir